### PR TITLE
Use public domain resources for examples.

### PR DESCRIPTION
**Pull request by MrZeusTheCoder**
* [X] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before? https://en.sfml-dev.org/forums/index.php?topic=13420
* [X] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [X] Have you provided some example/test code for your changes? Sort of. It is a bug fix.
* [X] If you have additional steps which need to be performed list them as tasks!
* [X] Modify or remove examples/asset_licenses.md. (Forgive the typo.)
* [ ] Answer any further questions.
----

## Description

This PR fixes some questions about non-free resources which are used in the examples for SFML. It was also mentioned that the "SFML Pong" example could be an issue, so I changed it to "SFML Tennis". 

A large number of resources were replaced, and all of the replacements are public domain. A list of the assets + their licenses/creators can be found in example/asset_licences.md (again, sorry for the typo, :P).

This PR is related to the issue  #1627 

## Tasks

* [X] Tested on Linux
* [ ] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?

Build the examples, and test them.

### CODE REVIEW

**Comment by LaurentGomila** [comment]
> non-free resources which are used in the examples for SFML

I'm surprised of this statement. An effort was made in SFML 2, to use only free resources. Since you're talking about the non-free resources inside the SFML SDK, I assume you have a list of them, together with their respective license?

**Comment by binary1248** [comment]
@LaurentGomila This is the classic case of "it's not free unless explicitly declared as such". Some distributions have very strict requirements on the packages they make available through their package repository because they don't want to have to deal with any kind of litigation.

Saying something is or should be freeware is not enough, there has to be an explicit notice from a credible source and if we use these resources we will also have to state where they came from and under what license they are offered.

The README for Sansation contains the following text:
```
Sansation - freeware font family
Version 1.31
Bernd Montag © 2011 - All Rights Reserved

This font family is freeware and can be used freely for personal and commercial purposes.
Although a PayPal donation in return is very much appreciated.

You may share this font on CDs, websites,... with the following restrictions:

	- Modification of the font files is only allowed for personal use - don´t distribute a modified version of the font files!
	- Do not rename the font files!
	- Do not sell the font files!
	- Do not pass the font files without this textfile!
	- Make sure you have downloaded the latest update from www.dafont.com for best optical results.

If you have further questions - please contact me.

berndmontag@gmx.de
```
This unfortunately doesn't pass as "free" in the context of FOSS.

`PONG` also seems to be a registered trademark: https://tsdr.uspto.gov/#caseNumber=76148525&caseType=SERIAL_NO&searchType=statusSearch

In my opinion we should really just replace the resources with ones that are obviously and indisputably free. Which resources these should be is mostly a matter of taste.

**Comment by MrZeusTheCoder** [comment]
binary1248 is absolutely right on. So 

> Which resources these should be is mostly a matter of taste.

and that taste is up to you guys. Let me know if you guys like or hate what I've done with it.

**Comment by nabijaczleweli** [comment]
For reference [here](https://salsa.debian.org/games-team/libsfml/-/blob/4c3d2e818979c69f9fc57c23c1eefd8137ea2c60/debian/copyright) are the resources Debian excludes, with [here](https://salsa.debian.org/games-team/libsfml/-/blob/4c3d2e818979c69f9fc57c23c1eefd8137ea2c60/debian/libsfml-doc.README.Debian) having some more reasoning:

```
Files excluded from Debian package
================
The Debian documentation package ships some samples that won't run out
of the box. This is because the data files aren't free (and for some
of them it's questionable if it's even allowed to distribute them). To
run these samples you might need to find replacements for the following
files:

  examples/android/app/src/main/assets/canary.wav
  examples/android/app/src/main/assets/orchestral.ogg
  examples/iOS/resources/canary.wav
  examples/iOS/resources/orchestral.ogg
  examples/opengl/resources/background.jpg
  examples/opengl/resources/texture.jpg
  examples/shader/resources/background.jpg
  examples/sound/resources/canary.wav
  examples/sound/resources/orchestral.ogg
  examples/win32/resources
  examples/win32/resources/image1.jpg
  examples/win32/resources/image2.jpg

The Debian source package does not ship any of the xcode templates since
they also include non-free resources like the ones above.
```

**Comment by MrZeusTheCoder** [comment]
@nabijaczleweli thanks. I can confirm that all those resources are replaced with public domain ones. (Including the XCode template ones)

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

File: examples/asset_licences.md
Comment (eXpl0it3r) [inline examples/asset_licences.md:1]: Can we use Tuffy V1.1, which is only about 18KiB in size? Also whenever possible, I'd try to link to the original author: http://tulrich.com/fonts/

File: examples/asset_licences.md
Comment (eXpl0it3r) [inline examples/asset_licences.md:2]: While we know now that you made this sound specifically for this, a few years down the line, it won't necessarily be clear who MrZeusTheCoder is and how the file got into the public domain. Have you published this clip somewhere that we could link to?
Also, could you export it at a lower bitrate than 1411kbps, trying not to enlarge the repository size with binary data too much.

File: examples/asset_licences.md
Comment (eXpl0it3r) [inline examples/asset_licences.md:3]: While it hurts to say that, but maybe it could be stored as JPG, so we don't add 600KiB of image data?

File: examples/asset_licences.md
Comment (eXpl0it3r) [inline examples/asset_licences.md:4]: Maybe play around a bit with the compression or size, so we don't have a 320% file size increase

File: examples/asset_licences.md
Comment (eXpl0it3r) [inline examples/asset_licences.md:8]: Again, we'd need a source to link to, otherwise we end up with the same situation as we have right now 😄

File: examples/asset_licences.md
Comment (eXpl0it3r) [inline examples/asset_licences.md:9]: Since this file is a markdown file, could you maybe add some styling to it?
For example make this an actual list, put the file path into inline code blocks, and putting the link behind the `here` text.
Or alternatively, you could make a table with columns file, license, author, url

File: examples/joystick/Joystick.cpp
Comment (eXpl0it3r) [inline examples/joystick/Joystick.cpp:226]: Why the color change?

File: examples/shader/Shader.cpp
Comment (eXpl0it3r) [inline examples/shader/Shader.cpp:445]: Why the different colors?

File: examples/tennis/Tennis.cpp
Comment (eXpl0it3r) [inline examples/tennis/Tennis.cpp:270]: Why the color change?

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]
Thanks for taking this up! 🙂 

- In order to keep the overall size of the repo as small as possible, we should try and reduce the file sizes of the newly added resources, especially the font, that is used multiple times.
- Awesome that you provided some audio yourself! In order for us to not end up in the same situation where we don't know the origin of some of the art or sounds, we need some reference to know that they are from you and which confirms that they are in the public domain

File: examples/asset_licences.md
Comment (MrZeusTheCoder) [reply examples/asset_licences.md:1]: Absolutely. I will switch it over to that. And yes, I will change that.

File: examples/asset_licences.md
Comment (MrZeusTheCoder) [reply examples/asset_licences.md:2]: Yep. I will look into doing that. I didn't really make it clear that I wasn't even sure if you guys wanted to keep the file, so I didn't put much work into it.

I very much will do that. Binaries, :disappointed: .

File: examples/asset_licences.md
Comment (MrZeusTheCoder) [reply examples/asset_licences.md:3]: Again, absolutely.

File: examples/tennis/Tennis.cpp
Comment (MrZeusTheCoder) [reply examples/tennis/Tennis.cpp:270]: To answer all three questions about the colour: I got a little carried away with the enhancements. My changes should have really been in two PRs. One for the copyright stuff, one for these possible enhancements. I might properly remove these and squish da commit.

File: examples/asset_licences.md
Comment (MrZeusTheCoder) [reply examples/asset_licences.md:9]: Refer to my other answer on whether this was even going to be a thing.

File: examples/tennis/Tennis.cpp
Comment (eXpl0it3r) [reply examples/tennis/Tennis.cpp:270]: I don't really mind if you change stuff slightly, but it's probably best if you mention it in the PR description, so it's clear that it was intentional. 🙂

**Comment by MrZeusTheCoder** [comment]
This should resolve all the size issues. I did fix up the markdown file, and left the few changes to the examples. They can be removed, or moved to a different PR. They are just some small things I thought could be improved as I went through the examples.

I would also ask you don't merge this PR quite yet as I found a few more assets that need to be resolved in `tools/..`.

**Comment by MrZeusTheCoder** [comment]
Ok, I think I got everything in place. It would be great if someone could test the XCode templates because I don't have access to a machine to try it on.

**Review by eXpl0it3r — CHANGES_REQUESTED** [review]

File: examples/asset_licenses.md
Comment (MrZeusTheCoder) [reply examples/asset_licenses.md:15]: Ok. Fixed it. Will push changes soon.

File: examples/sound/Sound.cpp
Comment (MrZeusTheCoder) [reply examples/sound/Sound.cpp:91]: I did. Got that fixed, thanks.

File: examples/joystick/Joystick.cpp
Comment (MrZeusTheCoder) [reply examples/joystick/Joystick.cpp]: I didn't try this with the new font before committing. Fixed that up, thanks.

**Comment by eliasdaler** [comment]
Can you please rebase your branch on master so it's easier to read? Right now the year changes in license make it hard to separate the actual content of this review from the fluff.

**Review by eXpl0it3r — APPROVED** [review]

**Comment by eXpl0it3r** [comment]
@eliasdaler in case you want to take another look 🙂

**Review by eliasdaler — APPROVED** [review]
Looks good to me. 
@eXpl0it3r after this patch is merged, we'll need to check that SFML docs on sfml-dev.org use non-free resources as well. E.g. [here](https://www.sfml-dev.org/tutorials/2.5/graphics-sprite.php) or [here](https://www.sfml-dev.org/tutorials/2.5/graphics-view.php).

**Comment by eXpl0it3r** [comment]
There seems to be some issue with the resources regarding the iOS build. Will try and look into it later.

> After this patch is merged, we'll need to check that SFML docs on sfml-dev.org use non-free resources as well
It's not really critical or essential there, since we're not packaging the page somewhere and fair-use still has its place, but if anyone wants to invest the time, they're free to do so. 🙂

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Pull request merged**
